dmoj/executors/RUST.py

```diff
@@ -56,15 +57,44 @@ def create_files(self, problem_id, source_code, *args, **kwargs):
         with open(self._file('Cargo.toml'), 'wb') as f:
             f.write(CARGO_TOML)
 
+    shared_target = None
+
+    @classmethod
+    def get_shared_target(cls):
+        if cls.shared_target is not None:
+            return cls.shared_target
+
+        cargo_dir = os.path.expanduser('~/.cargo')
+        collisions = 0
+        while True:
+            maybe_target = cargo_dir + f'/dmoj-shared-target-{collisions}'
+            try:
+                os.mkdir(maybe_target, mode=0o775)
+            except FileExistsError:
+                pass
+
+            dirfd = os.open(maybe_target, os.O_RDONLY | os.O_CLOEXEC)
```

Comment:
I think this should work and provide a little more validation 

```suggestion
            dirfd = os.open(maybe_target, os.O_PATH | os.O_DIRECTORY | os.O_CLOEXEC)
```

otherwise LGTM

Reply:
`O_PATH` doesn't work, but I added `O_DIRECTORY`.

If we use `O_PATH`, `flock` fails with EBADF.